packages/spec-api/src/types.ts

```diff
@@ -45,9 +45,21 @@ export type MockApiForHandler<Handler extends MockRequestHandler> =
   Handler extends KeyedMockRequestHandler<infer K> ? KeyedMockApi<K> : MockApi;
 
 export interface MockApi {
```

Comment:
I think here it would be better to use a discriminated type so you are not tied to the existing structure. 
Add `kind: "mock-server-api" | "api-definition"`
or whatever other name make sense

Reply:
Done